med-assist-backend/src/main/java/inc/evil/medassist/appointment/service/AppointmentServiceImpl.java

```diff
@@ -42,6 +49,18 @@ public void deleteById(String id) {
     @Override
     @Transactional
     public Appointment create(Appointment appointmentToCreate) {
+        LocalDate appointmentDate = appointmentToCreate.getAppointmentDate();
+        LocalTime startTime = appointmentToCreate.getStartTime();
+        LocalTime endTime = appointmentToCreate.getEndTime();
+        String doctorId = appointmentToCreate.getDoctor().getId();
+        Doctor doctor = doctorRepository.findByIdAndLock(doctorId);
```

Comment:
Decided to use this approach (in order to prevent overlapping appointments):
https://www.2ndquadrant.com/en/blog/postgresql-anti-patterns-read-modify-write-cycles/